assets/scripts/main.js

```diff
@@ -110,4 +110,14 @@
       $('.wcppec-checkout-buttons').hide();
     }
   });
+  
+  // Fixes missing anchor on elementor widget pagination.
+  // Must be combined with <a name="widget"></a> in title.
+  // https://github.com/elementor/elementor/issues/4703
+  $(document).ready(() => {
+    $('.elementor-widget-woocommerce-products').attr('id', $('.elementor-widget-woocommerce-products').attr('data-id'));
+    $('.elementor-widget-woocommerce-products .page-numbers a').each((i, a) => {
+      $(a).attr('href', $(a).attr('href') + '#' + $('.elementor-widget-woocommerce-products').attr('id'));
```

Comment:
The first argument (which you named `i`) should contain the full jQuery object of the link already, no?

In addition, there may be multiple product listing widgets on a single page, so you should find the closest to the link:
```suggestion
      $(a).attr('href', $(a).attr('href') + '#' + $(a).closest('.elementor-widget-woocommerce-products').attr('id'));
```